gramps/gui/viewmanager.py

```diff
@@ -1188,6 +1225,35 @@ def autobackup(self):
         """
         Backup the current family tree.
         """
+#************
+        # Before every backup Gramps must check disk space.
+        # Needed space = last_backup_file_size + 5%
+#************
+
+        backup_name = self.dbstate.db.get_dbname()
+        backup_ext = ".gramps"
+        path = config.get('database.backup-path')
+        bpath = sorted(os.listdir(path))
+        for n, bfile in enumerate(bpath):
+            if os.path.isfile(os.path.join(path, bfile)) and bfile.startswith(backup_name) and bfile.endswith(backup_ext):
+                blastfile = bfile
+
+        bfile_size = round(os.path.getsize(os.path.join(path, blastfile))/(1024))
```

Comment:
A utility function to return the largest backup size would be neater.

Reply:
I think about this - we must use biggest or last backup file.
In result - Gramps must use last backup file for active database.
Reason:
If user delete non used info from database -> new_database_size < old_database_size.
And Gramps create new_database_backup with small size.

Reply:
Yes.  It just needs to be put into a utility function in `gramps/gen/utils/file.py`.

Reply:
I test this code two day.
In result - backup_file_size must be calculated from active_database_size.
For SQLite3 - new database with Test71 MB have 4,1 MB backup file (> 7%).
If user create new Family tree and import to tree info from other base - Gramps must say to user:
1. You need least 'calculated backup_file_size' MB
or (easy way on 'try - except')
2. You cannot save your file. Clear your disk and try save again.